### ISSUE TITLE
Stile replacements and other edge cases

### ISSUE THREAD

**Issue by 10992-osm** [streetcomplete/StreetComplete#3188]
See also: #2991, #3141, \<issue to be submitted\>.

- Since the stile quest has been changed such that what was `stile=step_over` now must necessarily have a `material=stone` or `material=wood`, it might be helpful to provide a method of tagging the (very few) step-over stiles which are made of another material (metal, probably).  One could add an option for metal stiles on the main screen, but I think this won't be too helpful, as they are pretty rare ([only 207 worldwide](https://overpass-turbo.eu/s/1apE "Overpass Turbo query for metal step-over stiles"), compared to [16659 total step-over stiles](https://taginfo.openstreetmap.org/tags/stile=stepover "step-over stiles on taginfo)).  ([originally in this comment ](https://github.com/streetcomplete/StreetComplete/issues/2991#issuecomment-900432638) in #2991)
- If stiles are re-surveyed (see \<issue to be submitted\>), then it might be useful to provide more options for their replacements.  Currently, only `barrier=kissing_gate` is provided.  It [was suggested](https://github.com/streetcomplete/StreetComplete/issues/2991#issuecomment-899841013) that a stile might be replaced with a gap, a kissing gate, or another type of gate.  A kissing gate is probably the most common replacement for a stile, so the others could be hidden in the `Other answers…` menu (`The stile has been replaced by another sort of gate`, `The stile is no longer here`/`There is a gap`, `Can't say`).  Any report of a gap probably needs a note with a photo, and remote editing, but I'm not 100% sure.

I'm going to post a message on [Talk-GB] for opinions specifically about re-surveying stiles (combined with \<issue to be submitted\>.
I will ask about:
- what stiles are likely to be replaced with, and
- how a 'gap' should be mapped / responded to by StreetComplete.

I will post updates:
- when I've sent the message to the list, and
- when I seem to have received all the responses I'm going to get.

**Comment by 10992-osm** [comment]
The message has now been posted on [Talk-GB]:
[[Talk-GB] Removal of and changes to stiles](https://lists.openstreetmap.org/pipermail/talk-gb/2021-August/027488.html)

**Comment by westnordost** [comment]
To the first point: Note that Streetcomplete does not try to cover ALL possible options but only those that are reasonably common. Let's not lose our way here.  The rest is captured via notes.

**Comment by 10992-osm** [comment]
Since it has now been 7 days since I posted the [message on the [Talk-GB] mailing list](https://lists.openstreetmap.org/pipermail/talk-gb/2021-August/027488.html "Message on the [Talk-GB] mailing list about removal and changes to stiles"), I'll summarise the responses below, as far as they relate to replacements of stiles etc, see #3187 for any other responses:

- @matkoniecz [suggested on [Talk-GB]](https://lists.openstreetmap.org/pipermail/talk-gb/2021-August/027489.html) that rarer replacements for stiles can be handled by leaving notes.  He also suggests that a gap could be mapped with `barrier=entrance` (as I've used on [node 8479250734](https://osm.org/node/8479250734)).

**Comment by westnordost** [comment]
Okay, since this is @matkoniecz quest and he seems to be still invested into it, I'll just assign it to him so he should decide what to do. From my standpoint, adding steel stiles would be will not fix, because they are so rare.

**Comment by matkoniecz** [comment]
> Is that correct?

Yes - something where one can take square crop of just gap/gate and would be

- readable when image is tiny
- without visual clutter around it
- not misleading
- freely licensed (Geograph/Wikimedia Commons images are automatically OK)

**Comment by 10992-osm** [comment]
# Possible gap photos

1. [![Image of a fairly large gap in a stone wall, with a field on the other side.](https://user-images.githubusercontent.com/83511672/136988806-cd6ddb9e-0f92-4303-9c02-d3ee4a33a504.png)](https://www.geograph.org.uk/photo/2447664 "Geograph: A Gap in the Wall cc-by-sa/2.0 - © Colin Bews - geograph.org.uk/p/2447664")  CC-BY-SA 2.0, gap is quite large, so it's obvious, not sure how realistic it is for many situations.  Not sure how well it can be cropped to a square.
1. [![Image of a narrower gap in a drystone wall, with fields and low hills on the other side.  Long grass is growing partially in front of the gap, somewhat obscuring the view of the gap itself.](https://user-images.githubusercontent.com/83511672/136989701-ea5564f6-d9d1-4bfb-8927-804f9d038244.png)](https://www.geograph.org.uk/photo/5612820 "Geograph: A Gap in the Wall cc-by-sa/2.0 - © Neil Theasby - geograph.org.uk/p/5612820")  CC-BY-SA 2.0  The gap is perhaps more normal, not as wide and less smooth.  However, the view of the gap is slightly obscured by some of the grass.
1. [![Image of a stone-paved gap through a drystone wall.  It goes through into a green, grassy field of sheep, with trees on the horizon.  The gap is not very wide, but it is clear, and it is probably not narrow enough to stop sheep.](https://user-images.githubusercontent.com/83511672/136990610-dd01b9d7-0ad7-434f-ae14-08d225957efa.png)](https://www.geograph.org.uk/photo/5401039 "Geograph: Gap in dry-stone wall, Rachub cc-by-sa/2.0 - © Meirion - geograph.org.uk/p/5401039")  CC-BY-SA 2.0  This photo shows a gap through a drystone wall.  I think it is probably a good photo, since it is representative of normal gaps, particularly ones you might find when a stile has been removed, unlike the first one, which looks more consciously constructed.
1. [![Image of a gap in a wooden fence, with some slight overhang of the middle horizontal bar from the right-hand post by the gap.  On the other side, we can see a field filled with long grass, and in the distance, we can see the sea, a lighthouse, a slight hill with trees on, and what appears to be a square church tower.](https://user-images.githubusercontent.com/83511672/136996385-4472544a-3949-471f-a4c1-b4778ac2aae0.png)](https://www.geograph.org.uk/photo/978642 "Geograph: Gap in fence cc-by-sa/2.0 - © Evelyn Simak - geograph.org.uk/p/978642")  CC-BY-SA 2.0  This photo shows a gap in a wooden fence, with a field of long grass on the other side.  It seems typical of the gaps you find in wooden fences.  The gap is fairly obvious.
1. [![Image shows a a gap between to larger wooden fenceposts in a wire fence.  On both sides of the fence, we appear to be on upper-mountain grassland, and in the background, we can see some hills and mountains.  The grass is more yellow-brown than green, but not because it is dry.  There are no feature of note, like trees or buildings.](https://user-images.githubusercontent.com/83511672/136996897-3d3dde87-6d10-4cab-ba0d-ee451c8fe8db.png)](https://www.geograph.org.uk/photo/550211 "Geograph: Gap in the fence on Black Knowe Head cc-by-sa/2.0 - © Eileen Henderson - geograph.org.uk/p/550211") CC-BY-SA 2.0  This photo is not very clear, due to the nature of it being a wire fence.  I think, though have no evidence, that gaps are probably not as common in wire fences as in stone walls or wooden fences.

---

I think that my favourites here are number 3 and number 4.  If I find any more, I'll edit this to add them.  I will now look for some photos of gates.

**Comment by matkoniecz** [comment]
> 'Gap' might be better than 'opening'. Will this wording also be used on the other barrier quest?

yes
